incubator/elasticsearch/templates/configmap.yaml

```diff
@@ -13,22 +13,42 @@ data:
     
     node.data: ${NODE_DATA:true}
     node.master: ${NODE_MASTER:true}
+{{- if hasPrefix "5." .Values.image.tag }}
+    node.ingest: ${NODE_INGEST:true}
+{{- end }}
     node.name: ${HOSTNAME}
 
-    # see https://github.com/kubernetes/kubernetes/issues/3595
-    bootstrap.mlockall: ${BOOTSTRAP_MLOCKALL:false}
-
     network.host: 0.0.0.0
 
     cloud:
       kubernetes:
         service: ${SERVICE}
         namespace: ${KUBERNETES_NAMESPACE}
 
+{{- if hasPrefix "2." .Values.image.tag }}
```

Comment:
Again version specific. Add well-documented config options for these that do not rely on versions.